pkg/provenance.go

```diff
@@ -267,3 +280,60 @@ func verifyProvenanceName(name string) error {
 
 	return nil
 }
+
+// Note: see https://github.com/sigstore/cosign/blob/739947de3d0197fbaab926bd9b896963ebf47a19/pkg/providers/github/github.go.
+func getReusableWorkflowID() (string, error) {
+	url := os.Getenv(requestURLEnvKey) + "&audience=" + audience
+
+	req, err := http.NewRequest("GET", url, nil)
+	if err != nil {
+		return "", err
+	}
+
+	req.Header.Add("Authorization", "bearer "+os.Getenv(requestTokenEnvKey))
+	resp, err := http.DefaultClient.Do(req)
+	if err != nil {
+		return "", err
+	}
+	defer resp.Body.Close()
+
+	var payload struct {
+		// Count string `json:"count"`
+		Value string `json:"value"`
+	}
+
+	// Extract the value from JSON payload.
+	decoder := json.NewDecoder(resp.Body)
+	if err := decoder.Decode(&payload); err != nil {
+		return "", err
+	}
+
+	// This is a JWT token with 3 parts.
+	parts := strings.Split(payload.Value, ".")
+	if len(parts) != 3 {
+		return "", fmt.Errorf("invalid jwt token: found %d parts", len(parts))
+	}
+
+	content := parts[1]
+
+	// Base64-decode the content.
+	token, err := base64.RawURLEncoding.DecodeString(content)
+	if err != nil {
+		return "", fmt.Errorf("base64.RawURLEncoding.DecodeString: %w", err)
+	}
+
+	// Extract fields from JSON payload.
+	var oidc struct {
+		JobWorkflowRef string `json:"job_workflow_ref"`
+	}
+
+	if err := json.Unmarshal(token, &oidc); err != nil {
+		return "", fmt.Errorf("json.Unmarshal: %w", err)
+	}
+
+	if oidc.JobWorkflowRef == "" {
+		return "", fmt.Errorf("invalid job_workflow_ref: %v", token)
```

Comment:
does token contain sensitive info? any concerns to log?

Reply:
+1, remove and just replaced with `job_workflow_ref  is empty`

Reply:
now that I think of it, it was probably fine because I set the audience to `slsa-framework` so the token would not be accepted by anyone. But that's fine, this error should not fire up in practice